core/src/sumeragi/fault.rs

```diff
@@ -294,8 +262,133 @@ impl<F: FaultInjection> SumeragiWithFault<F> {
         }
         panic!("Timed out waiting for peers to come online");
     }
+
+    pub fn pipeline_time(&self) -> Duration {
+        self.block_time + self.commit_time
+    }
+}
+
+fn block_commit<F>(
+    sumeragi: &SumeragiWithFault<F>,
+    block: VersionedValidBlock,
+    state_machine_guard: &mut SumeragiStateMachineData,
+) where
+    F: FaultInjection,
+{
+    let block = block.commit();
+    let block_hash = block.hash();
+
+    if let Err(error) = state_machine_guard.wsv.apply(block.clone()) {
+        error!(%error);
+        panic!("Failed to apply block on WSV. This is absolutely not acceptable.");
+    }
+
+    // Update WSV copy that is public facing
+    {
+        let mut wsv_for_public_use_guard = sumeragi.wsv_for_public_use.lock().unwrap();
+        *wsv_for_public_use_guard = WorldStateView::clone(&state_machine_guard.wsv);
+    }
+
+    for event in Vec::<Event>::from(&block) {
+        trace!(?event);
+        drop(sumeragi.events_sender.send(event));
```

Comment:
These drops need a bit of explaining. I don't understand why they can't just go out of scope.

Reply:
They were inherited. Turns out the drops could just be removed.